install/install.go

```diff
@@ -167,6 +168,28 @@ type AzureParameters struct {
 	ClientSecret         string
 }
 
+var (
+	// DeprecatedFlagsToHelmOpts maps the deprecated install flags to the helm
+	// options
+	DeprecatedFlagsToHelmOpts = map[string]string{
+		"agent-image":              "image.override",
+		"azure-client-id":          "azure.clientID",
+		"azure-client-secret":      "azure.clientSecret",
+		"azure-resource-group":     "azure.resourceGroup",
+		"azure-subscription-id":    "azure.subscriptionID",
+		"azure-tenant-id":          "azure.tenantID",
```

Comment:
In my opinion, we should not deprecate those flag. It will add unnecessary cognitive load (what is helm? what is this helm-set doing? where is the reference for valid `helm-set` values?) to anyone trying out Cilium for the first time on Azure. 

With the PR as is, running `cilium install` on Azure will now also tell users to use a deprecated flag:
https://github.com/cilium/cilium-cli/blob/ad8d0d82d8ff16550bb9b05159fbe12999875cf7/install/azure.go#L115

I think this is a clear downgrade in user experience.

Reply:
Yes. I think having overlap is not necessarily a bad thing, if it helps with discovery of important options (such as those parameters  which every AKS user has to set).